File: src/main/java/com/timyr_tm/rust_bound/client/DebugEntries.java
package com.timyr_tm.rust_bound.client;

import com.timyr_tm.rust_bound.RustBound;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.components.debug.DebugScreenEntry;
import net.minecraft.resources.Identifier;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.neoforge.client.event.RegisterDebugEntriesEvent;
import net.neoforged.neoforge.client.event.RegisterDebugRenderersEvent;

import java.util.HashMap;
import java.util.Map;

@EventBusSubscriber(value = Dist.CLIENT, modid = RustBound.MOD_ID)
public final class DebugEntries {
	private static final Map<Identifier, DebugScreenEntry> ENTRY_MAP = new HashMap<>();

	public static final Identifier CONNECTION_ENDPOINTS = register(
		Identifier.fromNamespaceAndPath(RustBound.MOD_ID, "connection_endpoints"),
		new DebugEntryConnectionsEndpoint()
	);

	private static Identifier register(Identifier identifier, DebugScreenEntry entry) {
		ENTRY_MAP.put(identifier, entry);
		return identifier;
	}

	@SubscribeEvent
	private static void onRegisterDebugEntries(RegisterDebugEntriesEvent event) {
		for(Map.Entry<Identifier, DebugScreenEntry> entry : ENTRY_MAP.entrySet())
			event.register(entry.getKey(), entry.getValue());
	}

	@SubscribeEvent
	private static void onRegisterDebugRenderers(RegisterDebugRenderersEvent event) {
		if (Minecraft.getInstance().debugEntries.isCurrentlyEnabled(CONNECTION_ENDPOINTS))
			event.register(new DebugEntryConnectionsRender(Minecraft.getInstance()));
	}
}

File: src/main/kotlin/com/timyr_tm/rust_bound/client/debug/DebugEntries.kt
package com.timyr_tm.rust_bound.client.debug;

import com.timyr_tm.rust_bound.RustBound;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.components.debug.DebugScreenEntry;
import net.minecraft.resources.Identifier;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.neoforge.client.event.RegisterDebugEntriesEvent
import net.neoforged.neoforge.client.event.RegisterDebugRenderersEvent

@EventBusSubscriber(value = [Dist.CLIENT], modid = RustBound.MOD_ID)
object DebugEntries {
	private val ENTRY_MAP: MutableMap<Identifier, DebugScreenEntry> = mutableMapOf();

	val CONNECTION_ENDPOINTS: Identifier = register(
		Identifier.fromNamespaceAndPath(RustBound.MOD_ID, "connection_endpoints"),
		DebugEntryConnectionsEndpoint()
	);

	private fun register(identifier: Identifier, entry: DebugScreenEntry): Identifier {
        ENTRY_MAP[identifier] = entry;
		return identifier;
	}

	@SubscribeEvent
	fun onRegisterDebugEntries(event: RegisterDebugEntriesEvent) {
		for (entry in ENTRY_MAP)
			event.register(entry.key, entry.value)
	}

	@SubscribeEvent
	fun onRegisterDebugRenderers(event: RegisterDebugRenderersEvent) {
		if (Minecraft.getInstance().debugEntries.isCurrentlyEnabled(CONNECTION_ENDPOINTS))
			event.register(DebugEntryConnectionsRender(Minecraft.getInstance()));
	}
}
